### PR TITLE
New rest parameter properties error message

### LINKED ISSUE #8827: Give better message for rest parameter properties
If I write the following code:

```ts
class C {
    constructor(private ...x) {
    }
}
```

then I get the error `',' expected.`.

Instead, I should get an error like `A parameter property cannot be declared using a rest parameter.`

As motivation, for the following

```ts
class C {
    constructor(private {x}) {
    }
}
```

we currently give the error `A parameter property may not be a binding pattern.`

Additionally, in fixing this, the two should be consistent. Preferably, the new error for the latter should be `A parameter property may not be declared using a binding pattern.`

### PR DESCRIPTION
**Pull request by OrangeShark**
Fixes #8827 

Some baselines unrelated to the changes I made have been modified or even deleted. For example `Report an error when compiler-options input is empty object.errors.txt` was deleted but I can't seem to locate the test even in master.

### CODE REVIEW

**Comment by sandersn** [comment]
I noticed the extra, bogus baselines too. Just take them out of the PR for now while we figure out how they got there.

**Comment by OrangeShark** [comment]
@sandersn Okay, I took them out and updated the pull request.

File: tests/baselines/reference/destructuringParameterDeclaration4.errors.txt [hunk 1]
```diff
@@ -34,7 +36,11 @@ tests/cases/conformance/es6/destructuring/destructuringParameterDeclaration4.ts(
                 ~~~~~~~~~~~~~~~
 !!! error TS2370: A rest parameter must be of an array type.
     function a3(...b?) { }            // Error, can't be optional
+                    ~
```
Comment (sandersn) [inline tests/baselines/reference/destructuringParameterDeclaration4.errors.txt]: why wasn't this an error before? weird.

File: tests/baselines/reference/destructuringParameterDeclaration4.errors.txt [hunk 2]
```diff
@@ -64,8 +70,8 @@ tests/cases/conformance/es6/destructuring/destructuringParameterDeclaration4.ts(
     var temp = [1, 2, 3];
     class C {
         constructor(public ...temp) { }  // Error, rest parameter can't have accessibilityModifier
```
Comment (sandersn) [inline tests/baselines/reference/destructuringParameterDeclaration4.errors.txt]: can you make the comment more accurate (basically paraphrase your new error)

**Comment by sandersn** [comment]
Looks good except for a minor comment change in the baseline.
Plus there is the mystery of the previously-missing error message for `...arg?`. Maybe the parser change somehow fixed it in passing?

File: tests/baselines/reference/destructuringParameterDeclaration4.errors.txt (hunk 1 above)
Comment (OrangeShark) [inline tests/baselines/reference/destructuringParameterDeclaration4.errors.txt]: I checked the git history and they never had an error when they were added. Seems it has something to do with the other bogus baselines. They have always outputted the correct error in the compiler.

**Comment by OrangeShark** [comment]
I changed the comment. I also believe the reason for the previously missing error message for `...arg?` was because the error was from the checker while the previous error of having a parameter property for rest parameters was from the parser. Now that the error is discovered in the checker, it now shows other error messages from the checker. 

For example compiling this
```typescript
function a3(...b?) {}

class C {
    constructor(public ...temp) {}
}
```
only prints `error TS1005: ',' expected.` currently.

**Comment by sandersn** [comment]
:+1:

**Comment by sandersn** [comment]
@RyanCavanaugh or @DanielRosenwasser can you take a look at the wording of the new error messages? I want to get another pair of eyes besides mine.

**Comment by DanielRosenwasser** [comment]
This change makes me happy :100:. We're just waiting on the Travis Build before we merge right now.

**Comment by sandersn** [comment]
Thanks again, @OrangeShark.

**Pull request merged**